### PR TITLE
Support running with jdk17

### PR DESCRIPTION
**Pull request by ARTI1208**
Server changes: https://github.com/JetBrains/projector-server/pull/103

### CODE REVIEW

**Comment by SerVB** [comment]
Let's firstly ask @Avvessalom to review and test

**Comment by Avvessalom** [comment]
Please, firstly update JVM versions on all workflows

**Review by Avvessalom — APPROVED** [review]
LGTM

File: projector-agent-ij-injector/src/main/kotlin/org/jetbrains/projector/agent/ijInjector/IjAwtTransformer.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        // language=java prefix="class GraphicsEnvironment { private static GraphicsEnvironment createGE()" suffix="}"
+        """
+          {
+            Class prjClassLoaderClazz = ClassLoader.getSystemClassLoader().loadClass("${ProjectorClassLoader::class.java.name}");
+            ClassLoader loader = (ClassLoader) prjClassLoaderClazz
+              .getDeclaredMethod("getInstance", new Class[0])
+              .invoke(null, new Object[0]);
+            Class envClass = loader.loadClass("$PROJECTOR_GRAPHICS_ENVIRONMENT_NAME");
+            Object envInstance = envClass.getDeclaredMethod("getInstance", new Class[0]).invoke(null, new Object[0]);
+            return (java.awt.GraphicsEnvironment) envInstance;
```
Comment (SerVB) [inline projector-agent-ij-injector/src/main/kotlin/org/jetbrains/projector/agent/ijInjector/IjAwtTransformer.kt:60]: Am I right that we transform awt classes here? Looks like it's not a right place: this module is called `ij-injectror`, so code here should only transform IntelliJ.

Also it's not clear if it works when running a plain Swing app, not IntelliJ.

Comment (ARTI1208) [reply projector-agent-ij-injector/src/main/kotlin/org/jetbrains/projector/agent/ijInjector/IjAwtTransformer.kt:60]: > Looks like it's not a right place

Maybe just rename module to `projector-agent-injector`? Or do you think a new module should be created?

> Also it's not clear if it works when running a plain Swing app, not IntelliJ.

`override val loadingState: IdeState? = null` specifies that no IntelliJ loading state is required to run transformer, so yes, it works

Comment (SerVB) [reply projector-agent-ij-injector/src/main/kotlin/org/jetbrains/projector/agent/ijInjector/IjAwtTransformer.kt:60]: > Maybe just rename module to projector-agent-injector? Or do you think a new module should be created?

Well, I'm not sure. I would rather try to do it without transformations... But let's postpone it.

> override val loadingState: IdeState? = null specifies that no IntelliJ loading state is required to run transformer, so yes, it works

Cool, thanks!

**Review by SerVB — APPROVED** [review]

**Pull request merged**